### PR TITLE
Fix subscription retrieval.

### PR DESCRIPTION
**Pull request by Meruja**
Fix for https://github.com/wso2/product-apim/issues/8825
The issue was that we have marked the 'apiId 'param as required. The definition is used by the ValidationInInterceptor to validate the incoming calls.

A new param(apiId-Query) with required = false is used here.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 1]
```diff
@@ -7236,6 +7236,18 @@ parameters:
...
     type: string
 
+  # API Identifier
+  # Specified as part of the query string
+  apiId-Query:
+    name: apiId
+    in: query
+    description: |
+      **API ID** consisting of the **UUID** of the API.
+      The combination of the provider of the API, name of the API and the version is also accepted as a valid API I.
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:7246]:
```suggestion
      The combination of the provider of the API, name of the API and the version is also accepted as a valid API ID.
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 2]
```diff
@@ -3884,7 +3884,7 @@ paths:
         2. Retrieving subscriptions for a specific API.
         `GET https://127.0.0.1:9443/api/am/publisher/v1/subscriptions?apiId=c43a325c-260b-4302-81cb-768eafaa3aed`
       parameters:
-        - $ref: '#/parameters/apiId-Q'
+        - $ref: '#/parameters/apiId-Query'
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:3887]:
```suggestion
        - $ref: '#/parameters/apiId-Q-Opt'
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 3]
```diff
@@ -7236,6 +7236,18 @@ parameters:
     required: true
     type: string
 
+  # API Identifier
+  # Specified as part of the query string
+  apiId-Query:
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:7241]:
```suggestion
  apiId-Q-Opt:
```

Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Fixed with 6c3fe3f

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml (hunk 2 above)
Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Fixed with 6c3fe3f

**Pull request merged**